cassandra-mesos-executor/src/test/java/io/mesosphere/mesos/frameworks/cassandra/executor/BackupManagerTest.java

```diff
@@ -0,0 +1,191 @@
+        assertTrue(new File(backupDir, KEYSPACE + "/" + TABLE + "/data.db").isFile());
+    }
+
+    private void createCassandraDirs(String keyspace, String table, String snapshot) throws IOException {
+        File tableDir = new File(dataDir, keyspace + "/" + table + "-0");
+        File snapshotDir = new File(tableDir, "snapshots/" + snapshot);
+        assertTrue(snapshotDir.mkdirs());
+
+        Files.createFile(new File(snapshotDir, "data.db").toPath());
+    }
+
+    private void createBackupDirs(String keyspace, String table) throws IOException {
+        File tableDir = new File(backupDir, keyspace + "/" + table);
+        assertTrue(tableDir.mkdirs());
+        Files.createFile(new File(tableDir, "data.db").toPath());
+    }
+
+    private void delete(File file) {
+        if (file.isDirectory()) {
+            File[] children = file.listFiles();
+
+            if (children != null) {
+                for (File child: children) {
+                    delete(child);
+                }
+            }
+        }
+
+        assertTrue(file.delete());
+    }
+
+    class TestJmxConnect implements JmxConnect {
+        private String dataDir;
+        private List<String> invocations = new ArrayList<String>();
+
+        TestJmxConnect(String dataDir) {
+            this.dataDir = dataDir;
+        }
+
+        public List<String> getInvocations() { return Collections.unmodifiableList(invocations); }
+
+        @NotNull
+        @Override
+        public RuntimeMXBean getRuntimeProxy() {
+            throw new UnsupportedOperationException();
+        }
+
+        @NotNull
+        @Override
+        public StorageServiceMBean getStorageServiceProxy() {
+            return newStorageService();
+        }
+
+        @NotNull
+        @Override
+        public EndpointSnitchInfoMBean getEndpointSnitchInfoProxy() {
+            throw new UnsupportedOperationException();
+        }
+
+        @NotNull
+        @Override
+        public ColumnFamilyStoreMBean getColumnFamilyStoreProxy(@NotNull String keyspace, @NotNull String table) {
+            return newColumnFamilyStore();
+        }
+
+        @NotNull
+        @Override
+        public List<String> getColumnFamilyNames(@NotNull String keyspace) {
+            return Arrays.asList(TABLE);
+        }
+
+        @Override
+        public void close() throws IOException {}
+
+        private StorageServiceMBean newStorageService() {
+            return (StorageServiceMBean) Proxy.newProxyInstance(getClass().getClassLoader(), new Class[]{StorageServiceMBean.class},
+                    new InvocationHandler() {
+                        @Override
+                        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                            String name = method.getName();
+                            switch (name) {
+                                case "takeSnapshot":
+                                    invocations.add("takeSnapshot");
```

Comment:
This is an appropriate way to handle void return type in Proxy InvocationHandler implementation.
Just the same way as when you invoke void method via Reflection it will return null.